tools/link.py

```diff
@@ -2427,6 +2427,12 @@ def generate_traditional_runtime_html(target, options, js_target, target_basenam
     document.body.appendChild(script);
   }
 '''
+    # add required helper functions such as tryParseAsDataURI
+    for filename in ('arrayUtils.js', 'base64Utils.js', 'URIUtils.js'):
+      content = shared.read_and_preprocess(utils.path_from_root('src', filename))
+      script.inline = content + script.inline
```

Comment:
`script.inline` may be `None` here, I think? The old code guarded against that. I can't find where that is prevented in the new code.

Reply:
The live right above sets `script.inline` unconditionally so I think its fine.

Reply:
Ah, yes, the indentation confused me. This is indented the same as line 2413, even though there are unindented lines in between (which are JS code, not Python).